src/ui/layout/mct-tree.vue

```diff
@@ -430,28 +430,36 @@ export default {
 
             return scrollTopAmount >= treeStart && scrollTopAmount < treeEnd;
         },
+        getLowercaseObjectName(domainObject) {
+            let objectName;
+            if (!domainObject) {
+                return objectName;
+            }
+
+            if (domainObject.name) {
+                objectName = domainObject.name.toLowerCase();
+            }
+
+            if (domainObject.object && domainObject.object.name) {
+                objectName = domainObject.object.name.toLowerCase();
+            }
+
+            return objectName;
+        },
         sortNameAscending(a, b) {
             // sorting tree children items
-            if (!(a.name && b.name)) {
-                if (a.object.name.toLowerCase()
```

Comment:
@nikhilmandlik here's how it looked before ^

Reply:
yeah that looks funny to me, `domainObject` should not have `object` inside it

Reply:
@michaelrogers I know you worked on some things related to sorting in the tree - do you have context on this?

Reply:
I believe the intention was to support passing in a tree object.composition and therefore sort the children of the array which are not full domain objects. This may no longer be in use since the sort approach has changed some over time.

Reply:
In testing this locally, I added a folder in `My Items> Unnamed Folder` and there was a case that domainObject.name was not present on a domainObject but there was a domainObject.object.name property. So it would seem that we would need to catch that scenario, appears to only appear on tree item additions. Below is the example object, where the `domainObject.name` is missing.

`{
   id: "537073c7-fdc2-42a0-ada7-f876d042934c"
   isNew: false
   leftOffset: "36px"
   navigationPath: "/browse/mine/f83fb952-1a6f-475e-aada-3054d08d24ad/537073c7-fdc2-42a0-ada7-f876d042934c"
   object: Object
   objectPath: Array(3)
}`